tests/pertpy/test_distance_pseudobulk.py

```
@@ -0,0 +1,409 @@
+    )
+    adata = _make_adata(X, [4, 4, 4])
+
+    result = Distance(metric="pearson_distance").pairwise(adata, groupby="group").values
+    assert np.isnan(result[0, 1])
+    assert np.isnan(result[0, 2])
+    assert np.isfinite(result[1, 2])
+
+
+@pytest.mark.parametrize("metric", PSEUDOBULK_METRICS)
+def test_pseudobulk_single_group(metric: str) -> None:
+    """K=1 pairwise returns a 1×1 zero matrix; onesided returns a scalar zero."""
```

Ruff (0.15.14)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)